File: dsa-23au/java-dsa/jon-app/src/main/java/com/jon/app/CrimeData.java
package com.jon.app;

import java.util.Date;



public class CrimeData {
    private String Year;
    private String County;
    private String crimeType;
    private int indexCount;
    private float indexRate;
    private int violentCount;
    private float violentRate;
    private int propertyCount;
    private float propertyRate;
    private int firearmCount;
    private float firearmRate;
    private int population;

//Constructor
    public CrimeData(String Year, String County, String crimeType, int population, int indexCount, float indexRate, int violentCount, float violentRate, int propertyCount, float propertyRate, int firearmCount, float firearmRate ) {
    this.Year = Year;
    this.County = County;
    this.crimeType = crimeType; 
    this.population = population;
    this.indexCount = indexCount;
    this.indexRate = indexRate;
    this.violentCount = violentCount;
    this.violentRate = violentRate;
    this.propertyCount = propertyCount;
    this.propertyRate = propertyRate;
    this.firearmCount = firearmCount;
    this.firearmRate = firearmRate; 
    
    }

    //Getters 
    public String getYear() {
        return Year;
    }

    public String getCounty() {
        return County;
    }

    public String getCrimeType() {
        return crimeType;
    }

    public int getPopulation() {
        return population;
    }

    public int getIndexCount() {
        return indexCount;
    }
   
    public float getIndexRate() {
        return indexRate;
    }

    public int getViolentCount() {
        return violentCount;
    }

    public float getViolentRate() {
        return violentRate;
    }
    
    public int getPropertyCount() {
        return propertyCount;
    }

    public float getPropertyRate() {
        return propertyRate;
    }

    public int getFirearmCount() {
        return firearmCount;
    }

    public float getFirearmRate() {
        return firearmRate;
    }    
    //toString method for better logging and printing
    @Override 
        public String toString() {
            return "CrimeData{" +
                    "date='" + Date + '\'' +
                    ", county='" + County + '\'' +
                    ", crimeType='" + crimeType + '\'' +
                    ", population='" + population + '\'' +
                    ", indexCount='" + indexCount + '\'' +
                    ", indexRate='" + indexRate + '\'' +
                    ", violentCount='" + violentCount + '\'' +
                    ", violentRate='" + violentRate + '\'' +
                    ", propertyCount='" + propertyCount + '\'' +
                    ", propertyRate='" + propertyRate + '\'' +
                    ", firearmCount='" + firearmCount + '\'' +
                    ", firearmRate='" + firearmRate + '\'' +
                    '}';
    }
}

File: dsa-23au/Jon-AppFinal/crime_project/crime_data/crime_data.py
# crime_data.py
class CrimeData:
    def __init__(self, County, Year, Population, indexCount, indexRate, violentCount, violentRate, propertyCount, propertyRate, firearmCount, firearmRate):
        self.County = County
        self.Year = Year
        self.Population = Population
        self.indexCount = indexCount
        self.indexRate = indexRate
        self.violentCount = violentCount
        self.violentRate = violentRate
        self.propertyCount = propertyCount
        self.propertyRate = propertyRate
        self.firearmCount = firearmCount
        self.firearmRate = firearmRate

    def __str__(self):
        return f"CrimeData{{County='{self.County}', Year='{self.Year}', Population='{self.Population}', " \
               f"indexCount='{self.indexCount}', indexRate='{self.indexRate}', violentCount='{self.violentCount}', " \
               f"violentRate='{self.violentRate}', propertyCount='{self.propertyCount}', " \
               f"propertyRate='{self.propertyRate}', firearmCount='{self.firearmCount}', firearmRate='{self.firearmRate}'}}"

